src/types/memoryview.rs

```diff
@@ -0,0 +1,19 @@
+use crate::err::PyResult;
+use crate::{ffi, AsPyPointer, PyAny};
+
+/// Represents a Python `memoryview`.
+#[repr(transparent)]
+pub struct PyMemoryView(PyAny);
+
+pyobject_native_type_core!(PyMemoryView, pyobject_native_static_type_object!(ffi::PyMemoryView_Type), #checkfunction=ffi::PyMemoryView_Check);
+
+impl PyMemoryView {
+    /// Creates a new Python `memoryview` object from another Python object that
+    /// implements the buffer protocol.
+    pub fn from(src: &PyAny) -> PyResult<&PyMemoryView> {
```

Comment:
I think this could be a `TryFrom` impl to make this available in generic context?

Reply:
This is modeled after [PyByteArray::from](https://docs.rs/pyo3/latest/pyo3/types/struct.PyByteArray.html#method.from).

Reply:
Could we then add additional `TryFrom` impls for both cases?

Reply:
Added in https://github.com/PyO3/pyo3/pull/3514/commits/d4ed66fff0acbcf115a6511b6258cf1c6b1fc706